argus/shell.py

```diff
@@ -0,0 +1,261 @@
+                         argus `resources` directory
+    :param location: Where to save the resource
+    """
+    # pylint: disable=protected-access
+    base_url = ci._RESOURCES_LINK
+    base_url = base_url.rsplit("/", 1)[0]
+
+    url_resource = urlparse.urljoin(base_url,
+                                    resource_path)
+    _download_resource(url_resource, location)
+
+
+def _get_images(image_id, image_name):
+    """Return a list with all the id's."""
+    images_list = []
+    if image_id:
+        try:
+            mng = manager.APIManager()
+            images = mng.image_client.list_images()["images"]
+            for img_id in image_id:
+                img_name = ""
+                for image in images:
+                    if image["id"].lower() == img_id.lower():
+                        img_name = image["name"].lower()
+                        break
+                images_list.append((img_id, img_name))
+        finally:
+            mng.cleanup_credentials()
+    elif image_name:
+        try:
+            mng = manager.APIManager()
+            images = mng.image_client.list_images()["images"]
+            for img_name in image_name:
+                img_id = ""
+                for image in images:
+                    if image["name"].lower() == img_name.lower():
+                        img_id = image["id"].lower()
+                        break
+                images_list.append((img_id, img_name))
+        finally:
+            mng.cleanup_credentials()
+
+    # filter unknow images
+    for image in images_list:
+        if not image[0] or not image[1]:
+            images_list.remove(image)
+            print("[ERROR] Invalid image imformation id: {} - name: {}".format(
+                image[0], image[1]))
+
+    return images_list
+
+
+def _get_base_directory(directory):
+    """Return the path to the root directory for this run."""
+    if directory:
+        if not os.path.isdir(directory):
+            os.mkdir(directory)
+        directory = directory
+    else:
+        directory = tempfile.mkdtemp(prefix="argus-", dir="/tmp")
+    return os.path.abspath(directory)
+
+
+def _prepare_argument_parser():
+    """Prepare the argument parser."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-d", "--directory",
+                        help="Directory that will hold the required files.")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-i", "--image_id", action="append",
+                       help="Glance image-id to use for this run.")
+    group.add_argument("-n", "--image_name", action="append",
+                       help="Glance image-name to use for this run."
+                            "This is dangerous, be sure the name matches"
+                            " just one image.")
+    parser.add_argument("-f", "--flavor",
+                        help="The flavor id.")
+    parser.add_argument("-t", "--tests",
```

Comment:
You can leave a default value of `Scenario` to have all the scenarios run in case nothing is specified.